### PR TITLE
[4.0] Fix Issue of inaccessible calendar in articles if width is small

### PR DESCRIPTION
**Pull request by Arpit-24**
Pull Request for Issue #24114 .

### Summary of Changes

Overflow is made visible for the ul element wrapping the datepicker

### Testing Instructions

go to administrator=>content=>articles=>new=>publish tab
set width of tab less than 920 pixel
open the first calender

### Expected result

Full Accessible Calender

### Actual result

Partially accessible calender is now fixed

### Documentation Changes Required

None

Screenshot (1229)

After this change

### CODE REVIEW

**Comment by infograf768** [comment]
Not sure I understand this patch

`.mce-fullscreen`
is specific to TinyMCE and, **if necessary**,  should be added in `atum/scss/vendor/_tinymce.scss`

> Overflow is made visible for the ul element wrapping the datepicker

As far as I know, the calendar has a `<div>` container, not a `<ul>`

can you explain better?

**Comment by infograf768** [comment]
concerning the `mce-fullscreen`  it looks like some mixup with https://github.com/joomla/joomla-cms/pull/24334

**Comment by Arpit-24** [comment]
> concerning the `mce-fullscreen` it looks like some mixup with #24334

Yes, It was a mix with another issue fix. Apologies for the mistake.

And regarding why making overflow visible fixes the issue is. In the joomla-custom-elements.
The style for joomla-tab > ul is set to overflow-y:hidden which causes it to hide elements that exceed its bound.

In lower screen sizes the date picker element the div itself is going out of the size defined by ul element for its children thus causing it to be hidden.

**Comment by infograf768** [comment]
This is not working here:
Screen Shot 2019-03-25 at 10 13 40

**Comment by C-Lodder** [comment]
The calendar z-index also needs changing.

Please don't add your code to the `template.scss`. This file is supposed to be used as the base for importing all sub-scss files

**Comment by Arpit-24** [comment]
> The calendar z-index also needs changing.
> 
> Please don't add your code to the `template.scss`. This file is supposed to be used as the base for importing all sub-scss files

Updated the commit to _form.scss file

**Comment by Arpit-24** [comment]
> This is not working here:
> Screen Shot 2019-03-25 at 10 13 40

I was unable to replicate the issue. Could you give me some more details? It seems fixed for me.

My Configuration :
Windows 10 x64
Google Chrome Version 73.0.3683.86

**Comment by ghost** [comment]
@Puneet-Jain-18 please test.

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on b15ad3c9273e31f0a4fee33355ae52b0379d7ee6<br /><br />I tested the display of the calendar and it is now OK.

Remains an issue here when the window is small enough forces scroll bars to display (Firefox)
In that case we may need to scroll to see the calendar fully, BUT clicking on the scroll bars closes the calendar.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24336">issues.joomla.org/tracker/joomla-cms/24336</a>.</sub>

**Comment by infograf768** [comment]
scollbars_calendar

**Comment by Arpit-24** [comment]
> Remains an issue here when the window is small enough forces scroll bars to display (Firefox)
> In that case we may need to scroll to see the calendar fully, BUT clicking on the scroll bars closes the calendar.

Eliminated the need for horizontal scrolling. For vertical scrolling in mobiles if we scroll normally using touch then it doesn't close as emulated by chrome, but yes clicking the scrollbar will close it as is the expected browser behavior of taking the focus away from all the elements and giving the focus to body tag, and loosing focus from the input field will close the calendar picker.

File: administrator/templates/atum/scss/blocks/_form.scss [hunk 1]
```diff
@@ -104,3 +104,16 @@ legend {
...
 }
+
+// Allows date picker to be visible even when going out of the ul elements dimensions
+joomla-tab[view=accordion] {
+  ul {
+    overflow: visible;
+  }
+}
+
+@media screen and (max-width: 484px) {
```
Comment (C-Lodder) [inline administrator/templates/atum/scss/blocks/_form.scss]: Please use the appropriate variables for the media query

**Review by C-Lodder — CHANGES_REQUESTED** [review]

File: administrator/templates/atum/scss/blocks/_form.scss (hunk 1 above)
Comment (Arpit-24) [reply administrator/templates/atum/scss/blocks/_form.scss]: I tried using variables but as of now it seems that media queries don't support variables.

I tried using both var(--breakpoint-sm) as well as calc(var(--breakpoint-sm)) but none of them worked.

**Comment by C-Lodder** [comment]
I mean a SCSS variable (should already be defined), not CSS variables

**Comment by Arpit-24** [comment]
> I mean a SCSS variable (should already be defined), not CSS variables

Unfortunately, I couldn't find an already defined scss variable to use in the _variables.scss file. created a new one for use.

**Comment by C-Lodder** [comment]
We use the following breakpoints:

```
$grid-breakpoints: (
  xs: 0,
  sm: 576px,
  md: 768px,
  lg: 992px,
  xl: 1200px
);
```

So you'll want to use:

```
@include media-breakpoint-down(sm) {
  .js-calendar {
    right: 0;
  }
}
```

**Comment by Arpit-24** [comment]
> We use the following breakpoints:
> 
> ```
> $grid-breakpoints: (
>   xs: 0,
>   sm: 576px,
>   md: 768px,
>   lg: 992px,
>   xl: 1200px
> );
> ```
> So you'll want to use:
> 
> ```
> @include media-breakpoint-down(sm) {
>   .js-calendar {
>     right: 0;
>   }
> }
> ```

Updated

**Comment by Puneet-Jain-18** [comment]
I have tested this item :white_check_mark: successfully on e8acf35022cf906f93e361553ff87c991121b059<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24336">issues.joomla.org/tracker/joomla-cms/24336</a>.</sub>

**Comment by ghost** [comment]
@infograf768 please retest

**Comment by tushar33** [comment]
I have tested this item :red_circle: unsuccessfully on 4fd3b6cb6627f17351cef541d98bddf07ef6e7e0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24336">issues.joomla.org/tracker/joomla-cms/24336</a>.</sub>

**Comment by tushar33** [comment]
I have tested this item :red_circle: unsuccessfully on 4fd3b6cb6627f17351cef541d98bddf07ef6e7e0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24336">issues.joomla.org/tracker/joomla-cms/24336</a>.</sub>

**Comment by yasin-96** [comment]
I have tested this item :white_check_mark: successfully on 4fd3b6cb6627f17351cef541d98bddf07ef6e7e0<br /><br />After following the testing instructions I can confirm that the calender is full accessible.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24336">issues.joomla.org/tracker/joomla-cms/24336</a>.</sub>

**Comment by johndeutesfeld** [comment]
I have tested this item :white_check_mark: successfully on 4fd3b6cb6627f17351cef541d98bddf07ef6e7e0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24336">issues.joomla.org/tracker/joomla-cms/24336</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24336">issues.joomla.org/tracker/joomla-cms/24336</a>.</sub>

**Comment by roland-d** [comment]
@Arpit-24 While checking this issue I think the problem is already fixed without your changes. Can you check?

This is what I see without your changes:
image

**Comment by roland-d** [comment]
@Arpit-24 Never mind, I see the issue in fullscreen mode.

**Comment by roland-d** [comment]
@Arpit-24 Must have been cache because I cannot reproduce the issue again. So can you check if your change is still necessary?

**Comment by infograf768** [comment]
I agree with @roland-d 
I cannot reproduce the issue here in 4.0-dev

taking off rtc

**Comment by infograf768** [comment]
Moving to discussion<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24336">issues.joomla.org/tracker/joomla-cms/24336</a>.</sub>

**Comment by infograf768** [comment]
In fact calendar.js has been modified since April 2019.
I guess this can be closed.

**Comment by roland-d** [comment]
@Arpit-24 Thank you for your contribution, if this issue still exists, feel free re-open the issue and we can check again.

**Pull request closed without merging**